authority/provisioner/sign_ssh_options.go

```diff
@@ -301,8 +335,15 @@ func (v *sshCertificateValidityValidator) Valid(cert *ssh.Certificate) error {
 		return errors.Errorf("unknown ssh certificate type %d", cert.CertType)
 	}
 
-	// seconds
-	dur := time.Duration(cert.ValidBefore-cert.ValidAfter) * time.Second
+	// To not take into account the backdate, time.Now() will be used to
+	// calculate the duration if ValidAfter is in the past.
```

Comment:
Is this code path ever used in cases where the cert was not immediately previously generated for signing?

Reply:
It's always used in your case, but the duration can be defined by the user, this change avoids errors when the certificate duration exceeds the maximum duration because of the backdate.

For example, let's say the maximum is 24h and we request a 24h cert, but because of the backdate the validity will be 24h1m, but it will really be ~24h using the time of the CA, as part of the validity is ~1m ago in the past.